workspace/build_target.go

```diff
@@ -56,17 +57,36 @@ func (b BuildDependencies) ContainerList() []narwhal.ContainerDefinition {
 }
 
 func (bt BuildTarget) EnvironmentVariables(data runtime.RuntimeEnvironmentData) []string {
-	result := make([]string, 0)
+	envMap := make(map[string]string)
+
 	for _, property := range bt.Environment {
-		if _, _, ok := plumbing.SaneEnvironmentVar(property); ok {
-			interpolated, err := TemplateToString(property, data)
-			if err == nil {
-				result = append(result, interpolated)
-			} else {
-				result = append(result, property)
+		interpolated, err := TemplateToString(property, data)
```

Comment:
This seems to repeat the code from build_manifest.go quite a bit. Is there a way we can factor out a common function?

Reply:
OK!